src/plugins/data/public/search/session/session_service.ts

```diff
@@ -247,11 +261,21 @@ export class SessionService {
 
   /**
    * Infers search session options for sessionId using current session state
+   *
+   * In case user doesn't has access to `search-session` SO returns null,
+   * meaning that sessionId and other session parameters shouldn't be used when doing searches
+   *
    * @param sessionId
    */
   public getSearchOptions(
     sessionId: string
-  ): Required<Pick<ISearchOptions, 'sessionId' | 'isRestore' | 'isStored'>> {
+  ): Required<Pick<ISearchOptions, 'sessionId' | 'isRestore' | 'isStored'>> | null {
+    // in case user doesn't have permissions to search session, do not forward sessionId to the server
+    // because user most likely also doesn't have access to `search-session` SO
+    if (!this.hasAccessToSearchSessions) {
+      return null;
```

Comment:
Noticed that after this pr #89570 we track all searches in es and `trackId` would through in case a user doesn't have access to `search-session` SO.
To avoid this ensure we don't send `sessionId` to the server in case a user doesn't have access to work with sessions